cellmlmanip/model.py

```diff
@@ -709,3 +724,39 @@ def find_variable(self, search_dict):
             if matched:
                 matches.append(variable)
         return matches
+
+    def replace_equation(self, equation):
+        """
+        Searches the model for an equation naming the same variable in its LHS as the given ``equation``, and replaces
+        it with the new equation.
+
+        As with :meth:`add_equation()` the LHS must be either a variable symbol or a derivative, and all numbers and
+        variable symbols used in the equation must have been obtained from this model, e.g. via :meth:`add_number()`,
+        :meth:`add_variable()`, or :meth:`get_symbol_by_cmeta_id()`.
+
+        :param lhs: An LHS expression (either a symbol or a derivative).
```

Comment:
Why would you want to do that?

Reply:
Don't we need that for unit conversion?

Reply:
I think you would add a variable and then set the equation for that new variable with some method. Then after that you'd replace the equation for `dv/dt` with a ref to that variable...

Happy to talk API design though. We could have something like `set_equation(equation)` instead, which either sets an equation for the symbol in the LHS or replaces it if it already exists?

Reply:
Yes that might be a nice idea to have, as the former seems like quite a long winded way.

Reply:
Uhm... No you still couldn't do what you're suggesting above.
Creating variables is something different from setting equations.
Do you want a method that renames a variable?

Reply:
Could always just create variables if they don't already exist, but anway back to the main point.

`replace_equation` to me implies replacing one with another which isn't quite what this. `set_equation` might be better but it's also not quite intuitive as I wouldn't expect that to error if I created a new equation wiha lhs that hasn't currently got an eq. 
A simple solution would be to keep the name `replace_equation` and instead take 2 parameters and call them something like `old_lhs` and `new_rhs` instead of equation

Reply:
Every variable has a single _defining_ variable, we're replacing that one. The lhs might change in the method, for example you can replace `x = 1` with `dx/dt = 2`. So you can't write `old_lhs` because that might be different from the `new_lhs`. 

> I wouldn't expect that to error if I created a new equation wiha lhs that hasn't currently got an eq

Sorry I meant we could change this method to be called `set_equation` _and_ rewrite it a bit so that it can (1) add equations for variables that don't have one and (2) replace equations for variables that already have one.

Wrote an example at https://github.com/ModellingWebLab/cellmlmanip/issues/77

Reply:
I see I don't think that helps really.

I don't know what others think but 2 parameters to me makes more sense. Replace usually implies 2. I do see how they can both be wrapped into 1 Eq I find it a bit confusing, as it doesn't make clear that the lhs should be a pre-existing one. (your docstring doesn't mention that)

Also the docstring refers to lhs and rhs while there is just 1 parameter called equation so it's technically wrong and I doubt it'll play nicely with documentation generation

Reply:
Good point, I'll need to update the docstring! I don't mind to have `set_equation(lhs, rhs)` either

But why doesn't it help? You said you didn't expect a `set_` to throw errors if there's no eq for the variable in the lhs yet, so I'm going to make it stop throwing errors. What's still missing?

Reply:
exactly I wouldn't expect set_ to thow an error if the lhs doesn't already exist. Replace on the other hand that makse sense, so that's the better name. I'd do 'replace_equation(lhs, new_rhs)' or something like that

Reply:
I've changed it to `set_equation(lhs, rhs)` which can add or replace equations for the variable named in `lhs`

Reply:
That's more clear I think